app.py

```diff
@@ -135,35 +135,63 @@ def callback_view():
     return redirect(session.pop('next'))
   return redirect(url_for('demo_view'))
 
-@app.route('/hook/<pull_request_id>/<path:object_key>')
-def hook_view(pull_request_id, object_key):
-  repo_name = request.args.get('repo_name')
+def hook(args):
+  repo_name = args.get('repo_name')
   if not repo_name:
     return jsonify({'status': 'no repo name'})
   bot = bots.get(repo_name)
   storage = storages.get(repo_name)
   if bot:
-    if not pull_request_id.isdigit():
-      # pull_request_id is the branch name
+    if not args['pull_request_id'].isdigit():
+      # args['pull_request_id'] is the branch name
       if storage:
-        args = request.args.copy().to_dict()
         commit = args.pop('commit_id')
         if commit:
-          value = storage.get(pull_request_id, {})
+          value = storage.get(args['pull_request_id'], {})
           value[commit] = args
           value['current'] = args
-          storage.set(pull_request_id, value)
+          storage.set(args['pull_request_id'], value)
       try:
-        pull_request_id = bot.get_pr_by_branch(pull_request_id).number
+        args['pull_request_id'] = bot.get_pr_by_branch(args['pull_request_id']).number
       except:
         return jsonify({'status': 'no such pull request'})
-    url = url_for('go_view', object_key=object_key, _external=True)
-    if bot.process_hook(int(pull_request_id), url, request.args, storage):
+    url = url_for('go_view', object_key=args.get('object_key', ''), _external=True)
+    if bot.process_hook(int(args['pull_request_id']), url, args, storage):
       return jsonify({'status': 'success'})
     else:
       return jsonify({'status': 'restarting'})
   return jsonify({'status': 'no bot credentials'})
 
+@app.route('/hook/<pull_request_id>/<path:object_key>')
+def hook_view(pull_request_id, object_key):
+    args = request.args.copy().to_dict()
+    args.update({
+      'pull_request_id': pull_request_id,
+      'object_key': object_key,
+    })
+    return hook(args)
+
+@app.route('/coveralls', methods=['POST'])
```

Comment:
sure! happy to...but that would break existing installations, right? i don't see a version number anywhere (yet) that we could bump to indicate that...? up to you!